writer_test.go

```diff
@@ -151,6 +168,38 @@ func testWriterMaxAttemptsErr(t *testing.T) {
 	}
 }
 
+func testWriterMaxBytes(t *testing.T) {
+	topic := makeTopic()
+
+	createTopic(t, topic, 1)
+	w := newTestWriter(WriterConfig{
+		Topic:      topic,
+		BatchBytes: 25,
+	})
+	defer w.Close()
+
+	if err := w.WriteMessages(context.Background(), Message{
+		Value: []byte("Hi"),
+	}); err != nil {
+		t.Error(err)
+		return
+	}
+
+	if err := w.WriteMessages(context.Background(), Message{
+		Value: []byte("Hello World!"),
+	}); err == nil {
+		t.Error("expected error")
+		return
+	} else if err != nil {
+		switch err.(type) {
+		case MessageTooLargeError:
```

Comment:
I would love to see assertions for the `Message` and `Remaining` fields of the error.